pkg/controller/authenticationservice/standard.go

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2019, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package authenticationservice
+
+import (
+	"context"
+	adminv1beta1 "github.com/enmasseproject/enmasse/pkg/apis/admin/v1beta1"
+	"github.com/enmasseproject/enmasse/pkg/util"
+	"github.com/enmasseproject/enmasse/pkg/util/install"
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	resource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	intstr "k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func applyStandardAuthServiceDefaults(ctx context.Context, client client.Client, scheme *runtime.Scheme, authservice *adminv1beta1.AuthenticationService) error {
+	if authservice.Spec.Standard == nil {
+		authservice.Spec.Standard = &adminv1beta1.AuthenticationServiceSpecStandard{}
+	}
+	if authservice.Spec.Standard.Storage == nil {
+		authservice.Spec.Standard.Storage = &adminv1beta1.AuthenticationServiceSpecStandardStorage{
+			Type: adminv1beta1.Ephemeral,
+		}
+	}
+	if authservice.Spec.Standard.CertificateSecret == nil {
+		authservice.Spec.Standard.CertificateSecret = &corev1.SecretReference{
+			Name: "standard-authservice-cert",
+		}
+	}
+	if authservice.Spec.Standard.CredentialsSecret == nil {
+		authservice.Spec.Standard.CredentialsSecret = &corev1.SecretReference{
+			Name: "keycloak-credentials",
+		}
+		secret := &corev1.Secret{}
+		err := client.Get(ctx, types.NamespacedName{Namespace: authservice.Namespace, Name: "keycloak-credentials"}, secret)
+		if err != nil && errors.IsNotFound(err) {
+			secret.ObjectMeta = metav1.ObjectMeta{Namespace: authservice.Namespace, Name: "keycloak-credentials"}
+			install.ApplyDefaultLabels(&secret.ObjectMeta, "standard-authservice", "keycloak-credentials")
+			secret.StringData = make(map[string]string)
+			secret.StringData["admin.username"] = "admin"
+			secret.StringData["admin.password"] = util.GenerateAlphaNumeric(32)
+			if err := controllerutil.SetControllerReference(authservice, secret, scheme); err != nil {
+				return err
+			}
+			err := client.Create(ctx, secret)
+			if err != nil {
+				return err
+			}
+		}
+		return err
+	}
+	return nil
+}
+
+func applyStandardAuthServiceDeployment(authservice *adminv1beta1.AuthenticationService, deployment *appsv1.Deployment) error {
+
+	install.ApplyDeploymentDefaults(deployment, "standard-authservice", authservice.Name)
+
+	install.ApplyInitContainerWithError(deployment, "keycloak-plugin", func(container *corev1.Container) error {
```

Comment:
Ignoring the error will cause a warning in some IDEs. However there is a variant of this function `ApplyInitContainer` which does not require an error.

Reply:
Fixed